templates/admin/accounts/datacenter/fieldset.html

```diff
@@ -54,3 +55,54 @@
     {% endfor %}
 
 </fieldset>
+{% if fieldset.name == "Datacenter info" %}
+    
+    {% comment %} 
+        We have too many providers to list. Do not try to list them all. 
+    {% endcomment %}
+    {% if associated_providers_count > 1000 %}
+        <h2>Associated Hosting Providers</h2>    
+        <p>There are too many hosting providers using this datacentre to display.</p>
+        <p>If you need to see the full list, please 
+            <a href="https://www.thegreenwebfoundation.org/support-form/">
+            contact support.
+            </a>
+        </p>
+    {% endif %}
+
+    {% comment %} 
+        We have some providers listed in this DC, show them.
+    {% endcomment %}
+    {% if associated_providers_count < 1000 and associated_providers %}
```

Comment:
I noticed that this condition already exists in the view, and is repeated here in the template. What is your main concern about: the performance implications or how the list is displayed? Is there's a way to address the concern in a single place in code?

Reply:
Good point. I've updated this accordingly.